### DISCUSSION TITLE
GraphQL syntax highlighting is out of date

### DISCUSSION THREAD

**Discussion by aried3r** [github-linguist/linguist#6004] [Classification]
GraphQL syntax highlighting is currently based on the – now outdated – grammar defined here: https://github.com/rmosolgo/language-graphql

This breaks, i.e. when there are commas in the comments ([screenshot here](https://user-images.githubusercontent.com/1301152/183431283-101425fc-4bbc-4025-ba0d-16212b4f75fd.png), since, once fixed, this would not show the faulty behavior anymore):

```GraphQL
type Car {
  """
  This is fine
  """
  id: Int!

  """
  However, a comma can break syntax highlighting
  """
  speed: Int!
}

type Plane {
  id: Int!
  speed: Int!
}
```

Using https://github.com/graphql/graphiql/tree/main/packages/vscode-graphql-syntax this works in VS Code using the extension from that repository:

![Screenshot 2022-08-08 at 15 34 31](https://user-images.githubusercontent.com/1301152/183430651-a00d8c83-dee2-4e0a-a7d2-95ae2d5c096b.png)

Sadly I don't know how to vendor a more complex grammar like this for a PR. But I think using the official syntax highlighting grammar can be the way forward. WDYT?

**Comment by lildude** [comment]
If it fixes the problem and is more actively maintained, then I don't see any problems with switching grammars. You can find all the details on how to do that at https://github.com/github/linguist/blob/master/CONTRIBUTING.md#changing-the-source-of-a-syntax-highlighting-grammar .

**Reply by Bamblehorse** [reply]
Working on the above - will make a PR if I can get everything running 🙂 👍

**Reply by Bamblehorse** [reply]
@aried3r / @lildude

I have added a PR for the above: https://github.com/github/linguist/pull/6007

In line with the instructions here:
https://github.com/github/linguist/blob/master/CONTRIBUTING.md#changing-the-source-of-a-syntax-highlighting-grammar
